transcom/mymove
# INT B-18680 Departure Date For Origin Sit Is Not Persisting

## [B-18680](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a905470)

**NOTE:**
This PR includes other changes made by others that are in Main, but not in Integration.
These are the list of files that I have made changes to:

```
- pkg/handlers/primeapi/mto_service_item_test.go
- pkg/handlers/primeapi/payloads/payload_to_model.go
- pkg/handlers/primeapi/payloads/payload_to_model_test.go
- playwright/tests/office/txo/sitUpdates.spec.js
- src/components/Office/ConvertSITToCustomerExpenseModal/ConvertSITToCustomerExpenseModal.jsx
- src/components/Office/ReviewSITExtensionModal/ReviewSITExtensionModal.jsx
- src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.jsx
- src/components/Office/ShipmentSITDisplay/ShipmentSITDisplay.test.jsx
- src/components/Office/SubmitSITExtensionModal/SubmitSITExtensionModal.jsx
```

## Summary

When adding an Origin SIT via Prime and entering both entry and departure at the same time SIT is added, the departure date does not display in the SIT details > SIT Departure Date (TOO)

This PR fixes that and other related quick fixes were added as well.

**Updates:**

- Label changed: `SIT Departure Date` to `SIT departure date`.
- Date format changed to: `dd mm yyyy` to match.
- Value for `SIT departure date` now shows up in the SIT module when Prime requests a SIT and is approved by TOO.
- Added `SIT departure date` in Convert to Customer Expense modal to match with other views.

### How to test

1. Create a HHG move -> SC approves/submits -> TOO approves move
2. Find the move in Prime Simulator -> Click `Add Service Item` -> Fill out the form to add Origin SIT and make sure to include the SIT departure date.
3. Go to TOO's Move Task Order page -> Scroll down and approve SIT requested Items.
4. You should see the date for `SIT departure date` populated.
5. Also label should read `SIT departure date` with date format of `dd mm yyyy`.

## Screenshots
<img width="973" alt="image" src="https://github.com/transcom/mymove/assets/146856854/9ef02067-40c5-4335-8674-41bcc51fe4be">
<img width="718" alt="image" src="https://github.com/transcom/mymove/assets/146856854/9e0040f7-b0ac-48d2-b366-7cabcabd09b0">
